app/src/org/commcare/activities/RecoveryActivity.java

```diff
@@ -37,132 +37,141 @@
 
-                                if (result == FormUploadResult.FULL_SUCCESS) {
-                                    receiver.displayMessage(StringUtils.getStringRobust(
+                        if (result == FormUploadResult.FULL_SUCCESS) {
+                            updateStatus(StringUtils.getStringRobust(
+                                    RecoveryActivity.this,
+                                    R.string.recovery_forms_send_successful,
+                                    String.valueOf(successfulSends)));
+                            attemptRecovery();
+                        } else if (result == FormUploadResult.FAILURE) {
+                            String failureMessage = successfulSends > 0 ?
+                                    StringUtils.getStringRobust(
                                             RecoveryActivity.this,
-                                            R.string.recovery_forms_send_successful,
-                                            String.valueOf(successfulSends)));
-                                } else if (result == FormUploadResult.FAILURE) {
-                                    String failureMessage = successfulSends > 0 ?
-                                            StringUtils.getStringRobust(
-                                                    RecoveryActivity.this,
-                                                    R.string.recovery_forms_send_failure) :
-                                            StringUtils.getStringRobust(
-                                                    RecoveryActivity.this,
-                                                    R.string.recovery_forms_send_parital_success,
-                                                    String.valueOf(successfulSends));
-                                    receiver.displayMessage(failureMessage);
-                                } else if (result == FormUploadResult.TRANSPORT_FAILURE) {
-                                    receiver.displayMessage(StringUtils.getStringRobust(RecoveryActivity.this, R.string.recovery_forms_send_network_error));
-                                } else if (result == FormUploadResult.RECORD_FAILURE) {
-                                    receiver.displayMessage(Localization.get("sync.fail.individual"));
-                                } else if (result == FormUploadResult.AUTH_OVER_HTTP) {
-                                    receiver.displayMessage(Localization.get("auth.over.http"));
-                                }
-                            }
+                                            R.string.recovery_forms_send_failure) :
+                                    StringUtils.getStringRobust(
+                                            RecoveryActivity.this,
+                                            R.string.recovery_forms_send_parital_success,
+                                            String.valueOf(successfulSends));
+                            updateStatus(failureMessage);

```

Comment:
I believe all of these should be 

receiver.updateStatus() 

to handle rotation/re-attachment of tasks

Reply:
Spend some time looking at our configuration retention code and had an Aha moment! Guess all accessors to the activity's methods and variables need to be modified, and not just the `updateStatus` calls.

Reply:
Yeah, this is one of those frustrating Android things. The nice part about updateStatus type wrapper methods is that it prevents us from having to make repeated calls that are prefixed with the `receiver.` accessor, but basically we should treat the inside of deliverResult methods as if they were static and `receiver` is the only access it has to the surrounding context.